contracts/FeeDisbursal/FeeDisbursal.sol

```diff
@@ -109,129 +120,201 @@ contract FeeDisbursal is
+        Disbursal storage disbursal = _disbursals[token][proRataToken][index];
         snapshotId = disbursal.snapshotId;
         amount = disbursal.amount;
-        proRataToken = disbursal.proRataToken;
         recycleableAfterTimestamp = disbursal.recycleableAfterTimestamp;
         disburser = disbursal.disburser;
     }
 
     /// @notice get count of disbursals for given token
     /// @param token address of the claimable token
-    function getDisbursalCount(address token)
-    public
-    constant
-    returns (uint256)
+    /// @param proRataToken address of the token used to determine the user pro rata amount, must be a snapshottoken
+    function getDisbursalCount(address token, address proRataToken)
+        public
+        constant
+        returns (uint256)
     {
-        return _disbursals[token].length;
+        return _disbursals[token][proRataToken].length;
     }
 
-    /// @notice claim a token, to be called an investor
+    /// @notice accepts the token disbursal offer and claim offered tokens, to be called by an investor
     /// @param token address of the claimable token
-    /// @param until until what index to claim to
-    function claim(address token, uint256 until)
-    public
+    /// @param proRataToken address of the token used to determine the user pro rata amount, must be a snapshottoken
+    /// @param until until what index to claim to, noninclusive, use 2**256 to accept all disbursals
+    function accept(address token, ITokenSnapshots proRataToken, uint256 until)
+        public
     {
         // only allow verified and active accounts to claim tokens
-        require(_feeDisbursalController.onClaim(token, msg.sender), "");
-        claimPrivate(token, msg.sender, until);
+        require(_feeDisbursalController.onAccept(token, proRataToken, msg.sender), "NF_VERIFICATION_REQUIRED");
+        (uint256 claimedAmount, , uint256 nextIndex) = claimPrivate(token, proRataToken, msg.sender, until);
+
+        // do the actual token transfer
+        if (claimedAmount > 0) {
+            IERC223Token ierc223Token = IERC223Token(token);
+            assert(ierc223Token.transfer(msg.sender, claimedAmount, ""));
+        }
+        // log
+        emit LogDisbursalAccepted(msg.sender, token, proRataToken, claimedAmount, nextIndex);
     }
 
-    /// @notice claim multiple tokens, to be called an investor
+    /// @notice accepts disbursals of multiple tokens and receives them, to be called an investor
     /// @param tokens addresses of the claimable token
-    function claimMultiple(address[] tokens)
-    public
+    /// @param proRataToken address of the token used to determine the user pro rata amount, must be a snapshottoken
+    function acceptMultiple(address[] tokens, ITokenSnapshots proRataToken)
+        public
+    {
+        uint256[2][] memory claimed = new uint256[2][](tokens.length);
+        // first gather the funds
+        uint256 i;
+        uint256 totalAmount;
```

Comment:
We don't really need this, do we?

Reply:
according to designs we still need it. it will let you take all funds from single pro rata token (like NEUMARK) in one transaction. to me it makes sense. we still miss a method where you can take funds for single payment token but for multiple proRataTokens. then we save a lot of gas on transfers. I still plan to add it.